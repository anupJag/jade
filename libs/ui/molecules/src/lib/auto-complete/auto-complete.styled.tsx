import makeStyles from '@material-ui/core/styles/makeStyles';

export const useStyles = makeStyles(theme => ({
  listBox: {
    maxHeight: 500,
    overflowY: 'visible',
  },
  input: {
    flex: 1,
    fontSize: '16px',
    padding: '0 5px',
    color: '#778899',
  },
  image: {
    marginRight: theme.spacing(2),
    width: 50,
    height: 50,
  },
  img: {
    margin: 'auto',
    display: 'block',
    maxWidth: '100%',
    maxHeight: '100%',
  },
}));
