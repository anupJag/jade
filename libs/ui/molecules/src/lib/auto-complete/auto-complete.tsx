import React, { FC, useEffect } from 'react';
import Link from 'next/link';
import Autocomplete, { createFilterOptions } from '@material-ui/lab/Autocomplete';
import TextField from '@material-ui/core/TextField';
import ButtonBase from '@material-ui/core/ButtonBase';
import Hidden from '@material-ui/core/Hidden';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import parse from 'autosuggest-highlight/parse';
import match from 'autosuggest-highlight/match';
import { useStyles } from './auto-complete.styled';

interface AutocompleteProps {
  data: any[];
  onSuggestionChange: (suggestionText: string) => void;
  onSearchClick: (value: any) => void;
  placeholder?: string;
}

/**
 * filter configuration :
 *
 */
const filterOptions = createFilterOptions({
  matchFrom: 'start',
  trim: true,
});

const noop = () => true;

/**
 * Component : Autocomplete
 *
 * Customized auto complete component
 *
 * @param {AutocompleteProps} {
 *   data = [],
 *   onSuggestionChange = args => {},
 *   onSearchClick = args => {},
 * }
 * @returns
 */
export const AutoComplete: FC<AutocompleteProps> = ({
  data = [],
  onSuggestionChange = noop,
  onSearchClick = noop,
  placeholder = 'Search on store',
}: AutocompleteProps) => {
  const classes = useStyles();

  // useState : suggestions from props {data} will be stored
  const [suggestion, setSuggestion] = React.useState([]);

  // useState : onEnter / onClick of suggestions will updated here
  const [searchValue, setSearchValue] = React.useState(null);

  // useState : on change of input value will be updated here
  const [inputValue, setInputValue] = React.useState(null);

  // set suggestion
  useEffect(() => {
    setSuggestion(data);
  }, [data]);

  // Pass entered value to parent via callback
  useEffect(() => {
    onSearchClick(searchValue);
  }, [searchValue]);

  // Pass suggestion to parent via callback
  useEffect(() => {
    if (inputValue) onSuggestionChange(inputValue);
    else setSuggestion([]);
  }, [inputValue]);

  return (
    <Autocomplete
      id="search"
      freeSolo
      fullWidth
      autoComplete
      disableClearable
      disableListWrap
      clearOnEscape
      value={searchValue}
      options={suggestion}
      forcePopupIcon={false}
      filterOptions={filterOptions}
      classes={{ listbox: classes.listBox }}
      getOptionLabel={option => (typeof option === 'string' ? option : option.title)}
      onChange={(e, selctedValue) => setSearchValue(selctedValue)}
      onInputChange={(event: any, newInputValue: any) =>
        event?.type === 'change' && setInputValue(newInputValue)
      }
      renderInput={params => (
        <TextField
          {...params}
          InputProps={{
            ...params.InputProps,
            disableUnderline: false,
            className: classes.input,
          }}
          inputProps={{
            'aria-label': placeholder,
            ...params.inputProps,
          }}
          placeholder={placeholder}
          fullWidth
        />
      )}
      renderOption={option => {
        const matches = match(option.title, inputValue);
        const parts = parse(option.title, matches);
        return (
          <Grid container alignItems="center">
            <Hidden xsDown>
              <Grid item>
                <ButtonBase className={classes.image}>
                  <img
                    className={classes.img}
                    alt={option.images[0].altText}
                    src={option.images[0].url}
                  />
                </ButtonBase>
              </Grid>
            </Hidden>
            <Grid item xs>
              <Typography variant="body2" color="textPrimary">
                {parts.map((part, index) => (
                  <span key={index} style={{ fontWeight: part.highlight ? 700 : 400 }}>
                    {part.text}
                  </span>
                ))}
              </Typography>
            </Grid>
          </Grid>
        );
      }}
    />
  );
};
