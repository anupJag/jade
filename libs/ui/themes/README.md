# fe-themes

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test fe-themes` to execute the unit tests via [Jest](https://jestjs.io).
