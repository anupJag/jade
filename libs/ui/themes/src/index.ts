export * from './lib/defaultTheme';
export * from './lib/styled';
export * from './lib/themeProvider';
export { default as styled } from './lib/styled';
export * from './lib/GlobalStyles';
