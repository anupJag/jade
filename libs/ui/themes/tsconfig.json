{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "allowJs": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "types": ["node", "jest"]
  },
  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
}
