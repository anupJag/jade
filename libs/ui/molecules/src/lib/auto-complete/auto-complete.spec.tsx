import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { ThemeProvider } from '@jade/ui/themes';
import { AutoComplete } from './auto-complete';

describe('<AutoComplete />', () => {
  let AutoCompleteComponent;

  const onSuggestionChange = jest.fn();
  const onSearchClick = jest.fn();
  const data = [];
  const placeholder = 'Testing placeholder';
  let container;
  let inputEl;

  beforeEach(() => {
    AutoCompleteComponent = render(
      <ThemeProvider>
        <AutoComplete
          onSuggestionChange={onSuggestionChange}
          onSearchClick={onSearchClick}
          data={data}
          placeholder={placeholder}
        />
      </ThemeProvider>,
    );

    ({ container } = AutoCompleteComponent);
    inputEl = container.querySelector('input[type="text"]');
  });

  /**
   * Capturing Sanpshot : AutoComplete
   *
   *  To check style / structure changes in Styled Components
   */
  test('snapshot check', () => {
    const { asFragment } = AutoCompleteComponent;
    expect(asFragment()).toMatchSnapshot();
  });

  test('Search Input should have given placeholder', () => {
    const placeHolderTxt = inputEl.getAttribute('placeholder');
    expect(placeHolderTxt).toEqual(placeholder);
  });

  test('should respond on input click / enter', () => {
    fireEvent.click(inputEl);
    expect(onSearchClick).toHaveBeenCalled();
  });
});
