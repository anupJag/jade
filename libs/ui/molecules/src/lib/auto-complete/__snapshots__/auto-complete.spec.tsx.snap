// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutoComplete /> snapshot check 1`] = `
<DocumentFragment>
  <div
    aria-expanded="false"
    class="MuiAutocomplete-root MuiAutocomplete-fullWidth"
    role="combobox"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-2 MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
      >
        <input
          aria-autocomplete="both"
          aria-invalid="false"
          aria-label="Testing placeholder"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
          id="search"
          placeholder="Testing placeholder"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
